src/coreComponents/constitutive/solid/TriaxialDriver.cpp

```diff
@@ -0,0 +1,306 @@
+                              DomainPartition & GEOSX_UNUSED_PARAM( domain ) )
+{
+  // this code only makes sense in serial
+
+  GEOSX_THROW_IF( MpiWrapper::commRank() > 0, "Triaxial Driver should only be run in serial", std::runtime_error );
+
+  // get the solid out of the constitutive manager.
+  // for the moment it is of type SolidBase.
+
+  ConstitutiveManager & constitutiveManager = this->getGroupByPath< ConstitutiveManager >( "/Problem/domain/Constitutive" );
+
+  SolidBase & baseSolid = constitutiveManager.getGroup< SolidBase >( m_solidMaterialName );
+
+  // depending on logLevel, print some useful info
+
+  if( getLogLevel() > 0 )
+  {
+    GEOSX_LOG_RANK_0( "Launching Triaxial Driver" );
+    GEOSX_LOG_RANK_0( "  Material .......... " << m_solidMaterialName );
+    GEOSX_LOG_RANK_0( "  Type .............. " << baseSolid.getCatalogName() );
+    GEOSX_LOG_RANK_0( "  Mode .............. " << m_mode );
+    GEOSX_LOG_RANK_0( "  Strain Function ... " << m_strainFunctionName );
+    GEOSX_LOG_RANK_0( "  Stress Function ... " << m_stressFunctionName );
+    GEOSX_LOG_RANK_0( "  Steps ............. " << m_numSteps );
+    GEOSX_LOG_RANK_0( "  Output ............ " << m_outputFile );
+  }
+
+  // create a dummy discretization with one quadrature point for
+  // storing constitutive data
+
+  conduit::Node node;
+  dataRepository::Group rootGroup( "root", node );
+  dataRepository::Group discretization( "discretization", &rootGroup );
+
+  discretization.resize( 1 );   // one element
+  baseSolid.allocateConstitutiveData( discretization, 1 );   // one quadrature point
+
+  // set the initial stress state using the data table
+
+  arrayView3d< real64, solid::STRESS_USD > stressArray = baseSolid.getStress();
+
+  stressArray( 0, 0, 0 ) = m_table( 0, SIG0 );
+  stressArray( 0, 0, 1 ) = m_table( 0, SIG1 );
+  stressArray( 0, 0, 2 ) = m_table( 0, SIG2 );
+
+  baseSolid.saveConvergedState();
+
+  // pass the solid through the ConstitutivePassThru to downcast from the
+  // base type to a known model type.  the lambda here then executes the
+  // appropriate test driver. note that these calls will move data to device if available.
+
+  ConstitutivePassThru< SolidBase >::execute( baseSolid, [&]( auto & selectedSolid )
+  {
+    using SOLID_TYPE = TYPEOFREF( selectedSolid );
+
+    if( m_mode == "triaxial" )
+    {
+      runMixedControlTest< SOLID_TYPE >( selectedSolid, m_table );
+    }
+    else if( m_mode == "volumetric" || m_mode == "oedometer" )
+    {
+      runStrainControlTest< SOLID_TYPE >( selectedSolid, m_table );
+    }
+  } );
+
+  // move table back to host for output
+  m_table.move( LvArray::MemorySpace::CPU );
+
+  if( m_outputFile != "none" )
+  {
+    outputResults();
+  }
+
+  if( m_baselineFile != "none" )
+  {
+    compareWithBaseline();
+  }
+
+  return false;
+}
+
+
+void TriaxialDriver::outputResults()
+{
+  FILE * fp = fopen( m_outputFile.c_str(), "w" );
```

Comment:
Does that respect the `-o` flag in case of relative path provided?
I think we had some issues with @jhuang2601 with this.

Reply:
@TotoGaz Don't we have facilities to convert paths to absolute?

Reply:
Do you mean on the user side or on the developer side?
I don't know for both event if I imagine we have something on the dev side.

But here we directly dump the results directly in the string provided by the user; we should "_sanitize_" this on the developer side.